tests/acceptance/30_custom_promise_types/24_expireafter.cf

```diff
@@ -0,0 +1,86 @@
+######################################################
+#
+#  Test that expireafter works with custom promise types
+#
+#####################################################
+body common control
+{
+  inputs => { "../default.cf.sub" };
+  bundlesequence  => { default("$(this.promise_filename)") };
+  version => "1.0";
+}
+
+#######################################################
+
+bundle common version_check
+{
+  classes:
+      "python_version_compatible_with_cfengine_library"
+        expression => returnszero(
+          "/usr/bin/python3 -c 'import sys; assert sys.version_info >= (3,6)'",
+          "useshell"
+        );
+
+@if minimum_version(3.20)
+      "custom_promises_supports_expireafter"
+        expression => "any";
+@endif
+}
+
+bundle agent init
+{
+  meta:
+      "test_skip_unsupported"
+        string => "!custom_promises_supports_expireafter|!python_version_compatible_with_cfengine_library";
+
+  files:
+      "$(this.promise_dirname)/cfengine.py"
+        copy_from => local_cp(
+          "$(this.promise_dirname)/../../../modules/promises/cfengine.py"
+        );
+      "/tmp/testfile.txt"
+        delete => tidy;
+}
+
+#######################################################
+
+body action background {
+  background => "true";
+}
+
+bundle agent test
+{
+  meta:
+      "description" -> { "CFE-3435" }
+        string => "Test that expireafter works with custom promise types";
+
+  commands:
+      "$(sys.cf_agent) --inform --file $(this.promise_filename).sub --"
+        action => background;
+      "/bin/sleep 90";
+      "$(sys.cf_agent) --inform --file $(this.promise_filename).sub";
+      "/bin/sleep 100";
+}
```

Comment:
See my comment above; it's actually to allow the 1st agent run (backgrounded) enough time to finish sleeping and writing to file in the case where it wasn't correctly killed.

Reply:
See my comment below. I am confused why we are testing that the string promised by `slow_append` is in the file. I thought this test was making sure that the agent executing the first version of that was culled along with it's children processes so that it didn't continue to write.